nearcache_map.go

```diff
@@ -62,8 +84,52 @@ func newNearCacheMap(nc *inearcache.NearCache, ncc *nearcache.Config, ss *serial
 	return ncm, nil
 }
 
-func (ncm *nearCacheMap) registerInvalidationListener() {
-	fmt.Println("IMPLEMENT ME: registerInvalidationListener")
+func (ncm *nearCacheMap) Destroy(ctx context.Context, name string) error {
+	ncm.lg.Trace(func() string {
+		return fmt.Sprintf("hazelcast.nearCacheMap.Destroy: %s", name)
+	})
+	s := ncm.invalidationListenerID.Load()
+	if s == nil {
+		return nil
+	}
+	// removeNearCacheInvalidationListener
+	sid := s.(types.UUID)
+	ncm.rt.DeregisterHandler(name)
+	return ncm.lb.Remove(ctx, sid)
```

Comment:
Java client seems to destroy near cache here as well. https://github.com/hazelcast/hazelcast/blob/14ef0dbefa4a6fa00a50a8873aeea02ec6c3f5e4/hazelcast/src/main/java/com/hazelcast/client/map/impl/nearcache/NearCachedClientMapProxy.java#L582